### PR TITLE
Issue #891: Updated Layout styles (in progress)

### PR DESCRIPTION
**Pull request by wesruv**
Trying to give it some better visual hierarchy and use of white space
Fixed an issue with l-header in the layout area ~~and removed !importants~~

### CODE REVIEW

**Comment by quicksketch** [comment]
>  Fixed an issue with l-header in the layout area

I think the change to `l-header` may have impacts outside the layout editor. It's entirely possible to have a 2-column administrative page, or to be editing a 1-column layout in the layout editor. Will this change accidentally affect 2-column administrative pages?

**Comment by wesruv** [comment]
@quicksketch good point. You know I wasn't even prepared for you to merge this... I just put up a PR out of habit from other projects so people could see my code. Is that the a good way to go about that? If I had thought through it better I should have left a note in the description... Is there a better way to point people at a in progress edit?

Currently it's inadvertently the layout page, little hairy since the admin theme can have layouts inside of it. I'll see what I can see for a better solution.

**Comment by quicksketch** [comment]
> I just put up a PR out of habit from other projects so people could see my code. Is that the a good way to go about that?

Sure I think that's fine, but just say it's in-progress in the PR description. Also always please try to link back to the issue to which it's relevant by pasting in its URL (GitHub will shorten and associate the issue/PR that way). If it's relevant to a particular comment, paste in the URL to that comment (right click on the date/time to get the link).

> little hairy since the admin theme can have layouts inside of it. 

Yeah I think there's a selector, like `.layout-admin-edit` or something that you can use to target the wrapper around a layout that's being edited.

File: core/themes/seven/css/style.css [hunk 1]
```diff
@@ -367,13 +367,14 @@ ul.secondary li.active a.active {
   /* Add a margin to all layouts. */
   margin: 0 20px;
 }
-.l-header {
-  /* Then add a negative margin the header the same amount. */
+body > .layout > .l-header {
```
Comment (wesruv) [inline core/themes/seven/css/style.css:370]: @quicksketch how's this work for you? Should be general enough. Specificity is pretty high, but it shouldn't ever apply where we don't want it to :)

**Comment by wesruv** [comment]
@quicksketch Re: `.layout-admin-edit` the thing I worry about there is needing to get all of the layouts to behave a certain way, respecting `.layout-admin-edit`. 

Think it'll be a lot sturdier to focus on seven's header than individual layout's headers on the edit page.

**Comment by quicksketch** [comment]
> Think it'll be a lot sturdier to focus on seven's header than individual layout's headers on the edit page. 

Oh, I see what you're doing. I thought you wanted to target the header within the layout edit interface, but you specifically want to _not_ modify the layout edit interface...

I'm not sure about the child selectors, it seems too tied into the markup. Could we add a followup selector to just _undo_ the changes we don't want in the `.layout-admin-edit` selector? e.g.

``` css
.l-header {
  /* Styles */
}
.layout-admin-edit .l-header {
  /* Undo styles */
}
```

This approach probably has downsides as well. I'm just trying to figure out if it has fewer downsides...

**Comment by wesruv** [comment]
@quicksketch 
I think that seven would have to know specifics about any layout that would be inside the `.layout-admin-edit` area, which we couldn't do without scripting... 

If you didn't see what I was trying to get at I should probably point out that the negative margin on seven's header is currently affecting layouts in the `.layout-admin-edit area. If you look at the "before" screenshot on https://github.com/backdrop/backdrop-issues/issues/891 you'll see the header doesn't line up with anything because it's -15px off.

One thing just occured to me... why is seven (a theme) messing around with layout (which is now a separate 'thing')?

If we really want the contents of the header to be moved around a certain way but on in Seven:
- Could we add a class to all header regions with a preprocessor (or a layout module equivalent of a preprocessor)?
- Or could we put the negative margin on elements inside of the header that we want to line up a certain way?

**Comment by quicksketch** [comment]
> One thing just occured to me... why is seven (a theme) messing around with layout (which is now a separate 'thing')?

Yeah, there's still some intrinsic linking between the two at this point, especially when dealing with the margins around the outside of the page.

> Could we add a class to all header regions with a preprocessor (or a layout module equivalent of a preprocessor)?

Possibly... we don't have classes on regions currently though. Layout templates don't currently require classes, so adding them could be difficult from a compatibility standpoint.

> Or could we put the negative margin on elements inside of the header that we want to line up a certain way?

You mean the layout would provide the margins instead of the theme? I'm not sure this would work for us because Bartik and Seven can both use the single-column layout but have different margins in the header (and they do, currently).

You're probably right that child selectors is the best we can do. Browser support for `>` is complete, even for IE8. It's also unlikely to break within Seven because we control our own `page.tpl.php`.

**Comment by quicksketch** [comment]
This PR was closed after I merged into 1.x and deleted the 1.x-new-style branch. Could you refile against 1.x? Now that we're clear on child selectors being the best approach available (unless someone else has an idea), is this ready to go?

**Comment by wesruv** [comment]
@quicksketch I'm on it, probably won't get to this for a bit though. Have to get my talk ready for Dcon LA

**Comment by quicksketch** [comment]
Not a problem. We can release 1.1.0 as-is (once Pathauto is merged). You've done an amazing job! We can iterate and enhance after release. :smile:

**Comment by wesruv** [comment]
ooops wrong PR.

**Pull request closed without merging**